app/src/androidTest/java/ch/epfl/qedit/login/SignUpActivityTest.java

```diff
@@ -0,0 +1,225 @@
+            String password,
+            String passwordConfirmation) {
+        onView(ViewMatchers.withId(R.id.field_first_name))
+                .perform((typeText(firstName)))
+                .perform(pressImeActionButton());
+        onView(ViewMatchers.withId(R.id.field_last_name))
+                .perform((typeText(lastName)))
+                .perform(pressImeActionButton());
+        onView(ViewMatchers.withId(R.id.field_email))
+                .perform((typeText(email)))
+                .perform(pressImeActionButton());
+        onView(ViewMatchers.withId(R.id.field_password))
+                .perform((typeText(password)))
+                .perform(pressImeActionButton());
+        onView(ViewMatchers.withId(R.id.field_password_confirmation))
+                .perform((typeText(passwordConfirmation)))
+                .perform(pressImeActionButton());
+    }
+
+    private void performSignUp(
+            String firstName,
+            String lastName,
+            String email,
+            String password,
+            String passwordConfirmation) {
+        fillFields(firstName, lastName, email, password, passwordConfirmation);
+        clickOn(R.id.button_sign_up, true);
+    }
+
+    @SuppressWarnings("SameParameterValue")
+    private void testSignUpSuccessful(
+            String firstName,
+            String lastName,
+            String email,
+            String password,
+            String passwordConfirmation) {
+        performSignUp(firstName, lastName, email, password, passwordConfirmation);
+        //noinspection unchecked
+        intended(allOf(hasComponent(LogInActivity.class.getName())));
+    }
+
+    @SuppressWarnings("SameParameterValue")
+    private void testSignUpFailed(
+            String firstName,
+            String lastName,
+            String email,
+            String password,
+            String passwordConfirmation) {
+        performSignUp(firstName, lastName, email, password, passwordConfirmation);
+    }
+
+    private void performLogIn(String email, String password) {
+        onView(ViewMatchers.withId(R.id.field_email))
+                .perform((typeText(email)))
+                .perform(pressImeActionButton());
+        onView(ViewMatchers.withId(R.id.field_password))
+                .perform((typeText(password)))
+                .perform(pressImeActionButton());
+        clickOn(R.id.button_log_in, true);
+    }
+
+    @SuppressWarnings("SameParameterValue")
+    private void testLogInSuccessful(String email, String password, User user) {
+        performLogIn(email, password);
+        intended(allOf(hasComponent(HomeActivity.class.getName()), hasExtra(USER, user)));
+    }
+
+    private void testEmptyEntryCannotSignUp(int id) {
+        testWrongEntryCannotSignUp(id, "");
+    }
+
+    private void testWrongEntryCannotSignUp(int id, String stringToBeTyped) {
+        fillFields("Balboa", "Bark", "balboa.bark@river.tree", "honeyToast", "honeyToast");
```

Comment:
This is not a comment per-se, but I just admire Anthony's inspiration to find funny usernames and passwords. Maybe the great inspirations comes from later in the night.